src/core/MRApp.js

```diff
@@ -410,7 +424,37 @@ export class MRApp extends MRElement {
             this.stats.end();
         }
 
-        this.renderer.render(this.scene, this.user);
+        // ----- Actually Render ----- //
+
+        // TODO (in future) - once this gets more complicated, it will be nice to have a render system separate
+        // from the pure loop but it is okay as is here for now.
+
+        // Need to wait until we have all needed rendering-associated systems loaded.
+        if (this.maskingSystem == undefined) {
+            return;
+        }
+
+        const camera = this.user;
+
+        this.renderer.clear();
+
+        // Render panel to stencil buffer
+        this.renderer.state.buffers.stencil.setTest(true);
+        this.renderer.state.buffers.stencil.setMask(0xff);
+        this.renderer.render(this.scene, camera);
+
+        // Render child-objects to where the stencil buffer is set; must start at 1, see MaskingSystem for more details.
+        for (let panel_ref = 1; panel_ref <= this.maskingSystem.panels.length; ++panel_ref) {
+            this.renderer.state.buffers.stencil.setFunc(THREE.EqualStencilFunc, panel_ref, 0xff);
+            this.renderer.render(this.scene, camera);
```

Comment:
This is what's causing the hit. I tried commenting it out and got the FPS back, and masking still worked.

Reply:
looks like threejs is comin in clutch ~